Modules/cjkcodecs/multibytecodec.c

```diff
@@ -895,6 +901,76 @@ _multibytecodec_MultibyteIncrementalEncoder_encode_impl(MultibyteIncrementalEnco
     return encoder_encode_stateful(STATEFUL_ECTX(self), input, final);
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_getstate_impl(MultibyteIncrementalEncoderObject *self)
+/*[clinic end generated code: output=9794a5ace70d7048 input=4a2a82874ffa40bb]*/
+{
+    const char *pendingbuffer = NULL;
+    Py_ssize_t pendingsize;
+    MultibyteIncrementalEncoderState state;
+
+    memset(&state, 0, sizeof(state));
+    if (self->pending != NULL) {
+        pendingbuffer = PyUnicode_AsUTF8AndSize(self->pending, &pendingsize);
+        if (pendingbuffer == NULL) {
+            return NULL;
+        }
+        memcpy(state.pending, pendingbuffer, pendingsize);
+        state.pendingsize = pendingsize;
+    }
+    state.codecstate = self->state;
+
+    return _PyLong_FromByteArray((const unsigned char *)&state,
+                                 sizeof(state),
+                                 PY_LITTLE_ENDIAN, 0);
+}
+
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.setstate
+    statelong: object(type='PyLongObject *', subclass_of='&PyLong_Type')
```

Comment:
I feel `state` is better name than `statelong`.

Reply:
You can use `state as statelong` to use different name for argument name of Python and C.

Reply:
Thanks, "state as statelong" works nicely since there are two "state" variables here, but the user should not see "statelong".